src/foam/dao/ManyToManyRelationshipDAO.js

```diff
@@ -20,19 +20,42 @@ foam.CLASS({
   name: 'ManyToManyRelationshipDAO',
   extends: 'foam.dao.ProxyDAO',
 
-  implements: [ 'foam.mlang.Expressions' ],
+  implements: [ { path: 'foam.mlang.Expressions', java: false } ],
 
   documentation: 'Adapts a DAO based on a *:* Relationship.',
 
   properties: [
-    'junctionProperty',
-    'junctionDAOKey',
+    {
+      class: 'foam.core.Property',
+      javaType: 'foam.core.PropertyInfo',
+      name: 'junctionProperty'
+    },
+    {
+      class: 'foam.core.String',
+      name: 'junctionDAOKey'
+    },
     'junctionCls',
-    'targetProperty',
-    'sourceKey',
-    'sourceProperty',
+    {
+      class: 'foam.core.Property',
+      javaType: 'foam.core.PropertyInfo',
+      name: 'sourceKey'
+    },
+    {
+      class: 'foam.core.Property',
+      javaType: 'foam.core.PropertyInfo',
+      name: 'targetProperty'
+    },
+    {
+      class: 'foam.core.Property',
+      javaType: 'foam.core.PropertyInfo',
```

Comment:
Can you move all the javaType declarations to the java refinement class instead?